### PR TITLE
Add Ruff to pre-commit 

### PR DESCRIPTION
**Pull request by CalCraven**
This is just a test PR to test the linting with [Ruff](https://github.com/astral-sh/ruff), which is recommended by pydocstyle as a replacement and used by other big python package.

### CODE REVIEW

**Comment by CalCraven** [comment]
> If the only issue is the F401 errors in the __init__.py files then maybe we can just ignore that specific error in those files.

Yes that would be good. We may have to do that with a few of the different files this way because it's a bunch of files across the project.

**Comment by chrisjonesBSU** [comment]
Reading the ruff documentation https://docs.astral.sh/ruff/linter/#error-suppression

We can add `# ruff: noqa: F401` to the top of any `__init__.py` files which may be easier than setting the paths to ignore in a config file.

**Comment by CalCraven** [comment]
I would say we should ignore things for the `tests` module. Otherwise, I think it would be wise to perform the rest of the changes just to improper our PEP compliance.

**Comment by CalCraven** [comment]
https://github.com/BrianPugh/python-template/blob/main/.pre-commit-config.yaml

Going to link to a cookie-cutter version of a ruff linter, check out the pyproject.toml as well for some additional formatting stuff we could consider implementing, such as type checking and isort.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -8,28 +8,26 @@ ci:
     skip: []
     submodules: false
 repos:
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  # Ruff version.
+  rev: v0.4.2
+  hooks:
+    # Run the linter.
+    - id: ruff
+      args: [--line-length=80]
```
Comment (chrisjonesBSU) [inline .pre-commit-config.yaml:17]: Should we add the fix flag back in, so that easy formatting fixes are automatically made and comitted on PRs?

**Review by chrisjonesBSU — APPROVED** [review]
LGTM, just left one comment about how we want to handle automatic fixes on PRs.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (daico007) [reply .pre-commit-config.yaml:17]: yeah, sure! we can always turn it off later if it's become troublesome.

**Review by daico007 — APPROVED** [review]

**Pull request merged**